robot/impl/resource_manager.go

```diff
@@ -830,51 +830,60 @@ func (manager *resourceManager) updateResources(
 	defer manager.configLock.Unlock()
 	var allErrs error
 
-	for _, s := range conf.Added.Services {
+	seenServices := make(map[resource.Name]bool, 0)
+	for _, s := range append(conf.Added.Services, conf.Modified.Services...) {
```

Comment:
[highprio/bugprone] append will modify `conf.Added.Services` if it exceeds the capacity of the underlying slice. (https://go.dev/play/p/-lowOuyFfyS)

While I don't expect this to cause problems, it could lead to near-impossible to debug behavior

Reply:
ooh great catch! ill change